src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -409,7 +409,15 @@ ostree_repo_static_delta_execute_offline_with_signature (OstreeRepo *self, GFile
         return glnx_throw_errno_prefix (error, "openat(O_DIRECTORY)");
       else
         {
-          g_autofree char *dir = dirname (g_strdup (dir_or_file_path));
+          g_autofree char *dir = "";
```

Comment:
Initializing to `""` is wrong - that's a static buffer and we must not call `free()` on it. Set to `NULL` instead.

Reply:
Can we go back to the top here...I don't think the original code was wrong. But I think what we should do is simply:

```
g_autofree char *tmpbuf = g_strdup (dir_or_file_path);
const char *dir = dirname (tmpbuf);
```

Coverity should understand that (if it doesn't, it's arguably broken) and the intent is clearer.

Reply:
Thoughts on a previous review from @dbnicholson? https://github.com/ostreedev/ostree/pull/3263#discussion_r1638435360. Your suggested change is similar to my old one, but I don't have the second `g_autofree`, so will it solve the concern mentioned in the previous review?

Reply:
> Your suggested change is similar to my old one, but I don't have the second g_autofree, so will it solve the concern mentioned in the previous review?

Yes, not having the second `g_autofree` solves the concern.

Now, using `g_path_get_dirname()` sounded nice to me at the start, but being surprised by semantic differences is less nice.

So I definitely lean towards my suggestion above instead of `g_path_get_dirname` now.